src/microbe_stage/CompoundCloudSystem.cs

```diff
@@ -224,66 +186,44 @@ public void GetAllAvailableAt(Vector3 worldPosition, Dictionary<Compound, float>
         }
     }
 
-    /// <summary>
-    ///   Absorbs compounds from clouds into a bag
-    /// </summary>
     public void AbsorbCompounds(Vector3 position, float radius, CompoundBag storage,
         Dictionary<Compound, float>? totals, float delta, float rate)
     {
-        // It might be fine to remove this check but this was in the old code
         if (radius < 1.0f)
         {
             GD.PrintErr("Grab radius < 1 is not allowed");
             return;
         }
 
         int resolution = Resolution;
-
-        // This version is used when working with cloud local coordinates
         float localGrabRadius = radius / resolution;
 
-        float localGrabRadiusSquared = Mathf.Pow(radius / resolution, 2);
-
-        // Find clouds that are in range for absorbing
         foreach (var cloud in clouds)
         {
-            // Skip clouds that are out of range
             if (!cloud.ContainsPositionWithRadius(position, radius))
                 continue;
 
             cloud.ConvertToCloudLocal(position, out var cloudRelativeX, out var cloudRelativeY);
 
-            // Calculate all circle positions and grab from all the valid
-            // positions
-
-            // For simplicity all points within a bounding box around the
-            // relative origin point is calculated and that is restricted by
-            // checking if the point is within the circle before grabbing
             int xEnd = (int)Mathf.Round(cloudRelativeX + localGrabRadius);
             int yEnd = (int)Mathf.Round(cloudRelativeY + localGrabRadius);
 
-            // No lock needed here now as AbsorbCompounds now uses atomic reads and updates
             for (int x = (int)Mathf.Round(cloudRelativeX - localGrabRadius); x <= xEnd; x += 1)
             {
                 for (int y = (int)Mathf.Round(cloudRelativeY - localGrabRadius); y <= yEnd; y += 1)
                 {
-                    // Negative coordinates are always outside the cloud area
                     if (x < 0 || y < 0)
                         continue;
 
-                    // Circle check
-                    if (Mathf.Pow(x - cloudRelativeX, 2) + Mathf.Pow(y - cloudRelativeY, 2) >
-                        localGrabRadiusSquared)
-                    {
-                        // Not in it
+                    float distance = Mathf.Sqrt(Mathf.Pow(x - cloudRelativeX, 2) + Mathf.Pow(y - cloudRelativeY, 2));
+                    if (distance > localGrabRadius)
                         continue;
-                    }
 
-                    // Then just need to check that it is within the cloud simulation array
+                    float factor = 1.0f - (distance / localGrabRadius);
+
```

Comment:
What's the logic behind changes in this file? To me this seems like it first of all removes a ton of comments and seems to change the microbe compound absorption logic in a major way which is a game balance and performance concern as one of the most performance intensive parts of the game is all microbes absorbing compounds each game update (there's an added square root call per loop iteration which is a very heavy math operation).